containers-tests/tests/IntMapValidity.hs

```diff
@@ -9,57 +8,30 @@ import Utils.Containers.Internal.BitUtil (bitcount)
   Assertions
 --------------------------------------------------------------------}
 -- | Returns true iff the internal structure of the IntMap is valid.
-valid :: IntMap a -> Property
-valid t =
-  counterexample "nilNeverChildOfBin" (nilNeverChildOfBin t) .&&.
-  counterexample "commonPrefix" (commonPrefix t) .&&.
-  counterexample "maskRespected" (maskRespected t)
-
--- Invariant: Nil is never found as a child of Bin.
-nilNeverChildOfBin :: IntMap a  -> Bool
-nilNeverChildOfBin t =
-  case t of
-    Nil -> True
-    Tip _ _ -> True
-    Bin _ _ l r -> noNilInSet l && noNilInSet r
+valid :: IntMap a -> Bool
```

Comment:
`valid` seems too interesting to hide it in the tests.

@treeowl Would you be OK with moving it into the library?

Reply:
As long as it stays `Internal`, that seems reasonable.

Reply:
On the other hand, mightn't we eventually want to change its type to return a `Property` rather than a `Bool`, to explain *why* a tree is invalid? That would preclude inclusion in the library.

Reply:
OK. In any case it should be linked to a verbal description of the invariants in some way.